pandas/tests/frame/methods/test_reindex.py

```diff
@@ -83,6 +83,16 @@ class TestDataFrameSelectReindex:
     # These are specific reindex-based tests; other indexing tests should go in
     # test_indexing
 
+    def test_reindex_copies(self):
+        # based on asv time_reindex_axis1
+        N = 10
+        df = DataFrame(np.random.randn(N * 10, N))
+        cols = np.arange(N)
+        np.random.shuffle(cols)
+
+        result = df.reindex(columns=cols, copy=True)
```

Comment:
can you check with both axis reindexing as well (result should be the same) e.g. only a single copy.

Reply:
do you mean calling `df.reindex(index=foo, copy=True)`?  that's covered by existing tests.  this test should be focused on the particular bug

Reply:
yeah actually meant calling them *both* (columns AND index)

Reply:
updated + greenish